mmedit/models/backbones/sr_backbones/basicvsr.py

```diff
@@ -0,0 +1,384 @@
+        lrs_2 = lrs[:, :-1, :, :, :].reshape(-1, c, h, w)
+
+        flows_forward = self.spynet(lrs_1, lrs_2).view(n, t - 1, 2, h, w)
+        flows_backward = self.spynet(lrs_2, lrs_1).view(n, t - 1, 2, h, w)
+
+        return flows_forward, flows_backward
+
+    def forward(self, lrs):
+        """Forward function for BasicVSR.
+
+        Args:
+            lrs (Tensor): Input LR sequence with shape (n, t, c, h, w).
+
+        Returns:
+            Tensor: Output HR sequence with shape (n, t, c, 4h, 4w).
+        """
+        n, t, c, h, w = lrs.size()
+        assert h >= 64 and w >= 64, (
+            'The height and width of inputs should be at least 64, '
+            f'but got {h} and {w}.')
+
+        flows_forward, flows_backward = self.compute_flow(lrs)
+
+        # backward propgation branch
+        out_l = []
+        feat_prop = lrs.new_zeros(n, self.mid_channels, h, w)
+        for i in range(t - 1, -1, -1):
+            lr_curr = lrs[:, i, :, :, :]
+            if i < t - 1:  # no warping for the last timestep
+                flow = flows_backward[:, i, :, :, :]
+                feat_prop = flow_warp(feat_prop, flow.permute(0, 2, 3, 1))
+
+            feat_prop = torch.cat([lr_curr, feat_prop], dim=1)
+            feat_prop = self.backward_resblocks(feat_prop)
+
+            out_l.insert(0, feat_prop)
+
+        # forward propagation branch (including the upsampling module)
+        feat_prop = torch.zeros_like(feat_prop)
+        for i in range(0, t):
+            lr_curr = lrs[:, i, :, :, :]
+            if i > 0:  # no warping for the first timestep
+                flow = flows_forward[:, i - 1, :, :, :]
+                feat_prop = flow_warp(feat_prop, flow.permute(0, 2, 3, 1))
+
+            feat_prop = torch.cat([lr_curr, feat_prop], dim=1)
+            feat_prop = self.forward_resblocks(feat_prop)
+
+            # upsample given the backward and forward features
+            out = torch.cat([out_l[i], feat_prop], dim=1)
+            out = self.lrelu(self.fusion(out))
+            out = self.lrelu(self.upsample1(out))
+            out = self.lrelu(self.upsample2(out))
+            out = self.lrelu(self.conv_hr(out))
+            out = self.conv_last(out)
+            base = self.img_upsample(lr_curr)
+            out += base
+            out_l[i] = out
+
+        return torch.stack(out_l, dim=1)
+
+    def init_weights(self, pretrained=None, strict=True):
+        """Init weights for models.
+
+        Args:
+            pretrained (str, optional): Path for pretrained weights. If given
+                None, pretrained weights will not be loaded. Defaults to None.
+            strict (boo, optional): Whether strictly load the pretrained model.
+                Defaults to True.
+        """
+        if isinstance(pretrained, str):
+            logger = get_root_logger()
+            load_checkpoint(self, pretrained, strict=strict, logger=logger)
+        elif pretrained is None:
+            pass
```

Comment:
Here, the network will use the default initialization. Is it what you want? (I'm not sure.)

Reply:
Yes. Will change to
```
elif pretrained is not None:
    raise TypeError(...)
```